### ISSUE TITLE
Feature Request - Ideas : code generation

### ISSUE THREAD

**Issue by ichiriac** [Sevavietl/vscode-php-files#2]
I was thinking that it could be great to have other kind of features like :

- fully reformatting a file (I was looking for this one on vs-code and it's truly handy)
- be able to add new entities in currently opened file (with shortcuts for example)
- be able to detect the requested path and understand the namespace (like this you only need to declare the class name without the full namespace name)
- if you extends an abstract class, or implement an interface, automatically create needed methods

In order to do the last one you may need to know the structure of the project, I've started a project for this : https://github.com/glayzzle/php-reflection, but I'm thinking that the best way would be to provide a standalone reflection component and then expose it's api for others plugins in order to avoid parsing multiple times (that's why it's actually a work in progress)

**Comment by Sevavietl** [comment]
Yes, this extension really raw and only really basic naive functionality was added just to get going, so to speak.

I will try to do this:
"fully reformatting a file (I was looking for this one on vs-code and it's truly handy)"
Wasn't thinking about this, as all my files have only one entity. But it can be convenient.

I mostly implemented this:
"be able to detect the requested path and understand the namespace (like this you only need to declare the class name without the full namespace name)"

I really like this:
"if you extends an abstract class, or implement an interface, automatically create needed methods"
